packages/entity/src/__tests__/EntityQueryContext-test.ts

```diff
@@ -0,0 +1,102 @@
+import invariant from 'invariant';
+
+import { EntityQueryContext } from '../EntityQueryContext';
+import ViewerContext from '../ViewerContext';
+import { createUnitTestEntityCompanionProvider } from '../utils/testing/createUnitTestEntityCompanionProvider';
+
+describe(EntityQueryContext, () => {
+  describe('callbacks', () => {
+    it('calls all callbacks, and calls invalidation first', async () => {
+      const companionProvider = createUnitTestEntityCompanionProvider();
+      const viewerContext = new ViewerContext(companionProvider);
+
+      let preCommitFirstCallCount = 0;
+      let preCommitSecondCallCount = 0;
+      let postCommitCallCount = 0;
+      let postCommitInvalidationCallCount = 0;
+
+      const preCommitFirstCallback = async (): Promise<void> => {
+        preCommitFirstCallCount++;
+      };
+      const preCommitSecondCallback = async (): Promise<void> => {
+        preCommitSecondCallCount++;
+      };
+      const postCommitInvalidationCallback = async (): Promise<void> => {
+        postCommitInvalidationCallCount++;
+        invariant(
+          preCommitFirstCallCount === 1,
+          'preCommitInvalidation should be called before postCommits'
+        );
+        invariant(
+          preCommitSecondCallCount === 1,
+          'preCommitInvalidation should be called before postCommits'
+        );
+      };
+      const postCommitCallback = async (): Promise<void> => {
+        postCommitCallCount++;
+        invariant(
+          preCommitFirstCallCount === 1,
+          'preCommitInvalidation should be called before postCommits'
+        );
+        invariant(
+          preCommitSecondCallCount === 1,
+          'preCommitInvalidation should be called before postCommits'
+        );
+        invariant(
+          postCommitInvalidationCallCount === 1,
+          'postCommitInvalidation should be called before postCommit'
```

Comment:
```suggestion
          'postCommitInvalidation should be called before postCommits'
```

(or change "postCommits" to "postCommit" elsewhere)